addons/OPT/GPS/fn_addUnitToGPS.sqf

```diff
@@ -28,90 +28,96 @@
 #include "macros.hpp"
 
 
-params ["_newUnit", "_iconId"];
 
-DUMP("UNIT ICON ADDED");
-DUMP(_newUnit);
+[{
+    params ["_newUnit", "_iconId"];
+    DUMP("UNIT ICON ADDED");
+    DUMP(_newUnit);
 
-// Decide on the right color
-private _color = if (CLib_Player isEqualTo _newUnit) then {
-    COLOR_PLAYER_UNIT
-    } else {
-        if (group CLib_Player isEqualTo group _newUnit) then {
-            COLOR_OWN_GROUP
+    if (!([_newUnit] call FUNC(isUnitVisible))) exitWith {
+        DUMP(format ["Unit %1 is not visible", _newUnit]);
+    };
+    // Decide on the right color
+    private _color = if (CLib_Player isEqualTo _newUnit) then {
+        COLOR_PLAYER_UNIT
+        } else {
+            if (group CLib_Player isEqualTo group _newUnit) then {
+                COLOR_OWN_GROUP
+            } else {
+                COLOR_SIDE
+            }
+        };
+
+    private _specialTexture = getText (configFile >> "CfgVehicles" >> typeOf _newUnit >> "icon");
+    private _texture = format ["\A3\ui_f\data\map\vehicleicons\%1_ca.paa", if (_specialTexture == "") then {"iconMan"} else {_specialTexture}];
+    private _text = [_newUnit] call CFUNC(name);
+
+    if (_newUnit getVariable ["FAR_isUnconscious", 0] == 1) then {
+        _texture = "\A3\ui_f\data\igui\cfg\revive\overlayicons\u100_ca.paa";
+        if (_newUnit getVariable ["FAR_IsStabilized", 0] == 1) then {
+            _text = format ["%1 (%2)", [_newUnit] call CFUNC(name), "Stabilisiert" ];
+            _texture = "\A3\ui_f\data\igui\cfg\revive\overlayicons\u100_ca.paa";
+            _color = [0.850, 0.4, 0, 1];
```

Comment:
Konstanten für Farben lesen sich immer schöner, als Arrays :)

Reply:
Stimme ich dir zu, wird folgen :) . Ich träume noch von einer OPT-Farbpalette, die wir in allen Komponenten nutzen, zwecks einheitlichem Look...